test/code/plugins/VMInsightsDataCollector_subtest.rb

```diff
@@ -382,6 +382,8 @@ def test_get_filesystems_live
             @object_under_test = DataCollector.new
             @object_under_test.baseline
             actual = @object_under_test.get_filesystems
+            filesystems = actual.collect{|fs| fs.device_name}.to_set
+            omit "Running on container" if expected.size != actual.size && filesystems.size == expected.size
```

Comment:
This doesn't make sense to me. Assume there is a bug in the product code and it is **not** running in a container. Because of the bug, no file systems are returned in actual. The if condition will simply cause the test case to be omitted when it should be failing.

The decision to omit a test case must not depend upon the "actual" data. It must be be determined based upon the conditions found before exercising the code under test.